playbooks/archivematica-bionic/bionic-ci.jenkinsci

```diff
@@ -0,0 +1,176 @@
+            -f=json -o=results/output-uuid.json \
+            --no-skipped || true
+
+          env/bin/python -m behave2cucumber  -i results/output-uuid.json -o results/cucumber-uuid.json || true
+'''
+}
+
+stage('Run mo-aip-reingest') {
+sh '''
+export DISPLAY=:45
+timeout 30m env/bin/behave \
+          --tags=mo-aip-reingest \
+          --no-skipped \
+          -D am_version=1.7 \
+          -D driver_name=Firefox \
+          -D am_username=admin \
+          -D am_password=archivematica \
+          -D am_url=http://${SERVER}/ \
+          -D ss_username=admin \
+          -D ss_password=archivematica \
+          -D ss_url=http://${SERVER}:8000/ \
+          -D home=${USER} \
+          -D server_user=${USER} \
+          -D transfer_source_path=${USER}/archivematica-sampledata/TestTransfers/acceptance-tests \
+          -D ssh_identity_file=${KEY} \
+           --junit --junit-directory=results/ -v \
+            -f=json -o=results/output-moaip.json \
+            --no-skipped || true
+
+          env/bin/python -m behave2cucumber  -i results/output-moaip.json -o results/cucumber-moaip.json || true
+'''
+}
+
+stage('Run premis-events') {
+sh '''
+export DISPLAY=:45
+timeout 30m env/bin/behave \
+          --tags=premis-events \
+          --no-skipped \
+          -D am_version=1.7 \
+          -D driver_name=Firefox \
+          -D am_username=admin \
+          -D am_password=archivematica \
+          -D am_url=http://${SERVER}/ \
+          -D ss_username=admin \
+          -D ss_password=archivematica \
+          -D ss_url=http://${SERVER}:8000/ \
+          -D home=${USER} \
+          -D server_user=${USER} \
+          -D transfer_source_path=${USER}/archivematica-sampledata/TestTransfers/acceptance-tests \
+          -D ssh_identity_file=${KEY} \
+           --junit --junit-directory=results/ -v \
+            -f=json -o=results/output-premis.json \
+            --no-skipped || true
+
+          env/bin/python -m behave2cucumber  -i results/output-premis.json -o results/cucumber-premis.json || true
+'''
+}
+
+stage('Run icc,tpc,picc') {
+sh '''
+export DISPLAY=:45
+timeout 30m env/bin/behave \
+          --tags=icc,tpc,picc \
+          --no-skipped \
+          -D am_version=1.7 \
+          -D driver_name=Firefox \
+          -D am_username=admin \
+          -D am_password=archivematica \
+          -D am_url=http://${SERVER}/ \
+          -D ss_username=admin \
+          -D ss_password=archivematica \
+          -D ss_url=http://${SERVER}:8000/ \
+          -D home=${USER} \
+          -D server_user=${USER} \
+          -D transfer_source_path=${USER}/archivematica-sampledata/TestTransfers/acceptance-tests \
+          -D ssh_identity_file=${KEY} \
+           --junit --junit-directory=results/ -v \
+            -f=json -o=results/output-icc.json \
+            --no-skipped || true
+
+          env/bin/python -m behave2cucumber  -i results/output-icc.json -o results/cucumber-icc.json || true
+'''
```

Comment:
After reading these stages, I think it's starting to make sense to have a script with parameters to avoid some repetition.

Reply:
This stage can be run using only one step, that runs all the tags. I did them separately, so the process could be followed in smaller chunks, but that's not so useful once we reach production.

What I have in mind to avoid command repetition is something like:

```
TAGS=${TAGS:-@am17}
for i in $TAGS
   do
    behave ....
    done
```

This way we would be able to run specific tasks from the jenkins interface, and use a sane default when nothing is set

Reply:
Regarding parameters, I think we should update the command to pass the am/ss branches to deploy, and rely on the `vars-singlenode.yml` for all others, and if we need different vars for the qa/ branches, we can create a qa/1.x branch in deploy-pub that would be merged to master as part of the release.

This way, the same Jenkins job works for stable,qa and dev/ branches.